src/interfaces/IDCARegistry.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {ResolvedOrder} from "../base/ReactorStructs.sol";
+
+/// @notice Interface for DCA registry that tracks and validates DCA order execution
+interface IDCARegistry {
+    /// @notice DCA execution state for tracking chunks
+    struct DCAExecutionState {
+        uint256 executedChunks;
+        uint256 lastExecutionTime;
+        uint256 totalInputExecuted;
+        uint256 totalOutputReceived;
+    }
+
+    /// @notice Public DCA intent that users sign over
+    struct DCAIntent {
+        address inputToken; // Token to sell
+        address outputToken; // Token to buy
+        address cosigner; // Trusted cosigner address
+        uint256 minFrequency; // Minimum time between chunks (seconds)
+        uint256 maxFrequency; // Maximum time between chunks (seconds)
+        uint256 minChunkSize; // Minimum input amount per chunk
+        uint256 maxChunkSize; // Maximum input amount per chunk
+        uint256 minOutputAmount; // Minimum output amount expected for this DCA execution
+        uint256 maxSlippage; // Max slippage in basis points (10000 = 100%)
+        uint256 deadline; // Intent expiration timestamp
+        uint256 nonce; // Replay protection nonce
+        bytes32 privateIntentHash; // Hash of private parameters (total amount, chunks, etc)
```

Comment:
User's commitment to the private dca params; since it's only the hash, smart contract will not be able to validate against them

maybe there can be an auxiliary dispute system where our TEE gets slashed if it creates orders that violate those constraints, and user choose to reveal the preimage?